Sdk/Speckle.Connectors.Common/Operations/SendProgress.cs

```diff
@@ -0,0 +1,44 @@
+using Speckle.InterfaceGenerator;
+using Speckle.Sdk.Transports;
+
+namespace Speckle.Connectors.Common.Operations;
+
+[GenerateAutoInterface]
+public class SendProgress(IProgressDisplayManager progressDisplayManager) : ISendProgress
+{
+  private string? _previousSpeed;
+
+  public void Begin() => progressDisplayManager.Begin();
+
+  public void Report(IProgress<CardProgress> onOperationProgressed, ProgressArgs args)
+  {
+    if (args.ProgressEvent == ProgressEvent.UploadBytes)
+    {
+      switch (args.ProgressEvent)
+      {
+        case ProgressEvent.UploadBytes:
+          _previousSpeed = progressDisplayManager.CalculateSpeed(args);
+          break;
+      }
+    }
+    if (!progressDisplayManager.ShouldUpdate())
+    {
+      return;
+    }
+
+    switch (args.ProgressEvent)
+    {
+      case ProgressEvent.CachedToLocal:
+        onOperationProgressed.Report(new($"Caching... ({args.Count} total objects)", null));
```

Comment:
not sure cache reporting is a valueable info for user. I wouldn't report it as progress.